### PR TITLE
GLTFLoader  - Support Host Relative URL

### PR DESCRIPTION
**Pull request by gilnoy**

### CODE REVIEW

File: examples/js/loaders/GLTFLoader.js [hunk 1]
```diff
@@ -1246,6 +1246,11 @@ THREE.GLTFLoader = ( function () {
 
 		// Invalid URL
 		if ( typeof url !== 'string' || url === '' ) return '';
+		
+		// Host Relative URL
+		if (/^https?:\/\//i.test(path) && /^\//.test(url)) {
```
Comment (donmccurdy) [inline examples/js/loaders/GLTFLoader.js]: Could you give an example of a URL (or a model) with the issue you’re trying to solve? Something like `/texture.png`? This seems to be testing for URLs that begin with both `http` and `/` which would be mutually exclusive.

Comment (donmccurdy) [reply examples/js/loaders/GLTFLoader.js]: Oh, I missed that this was testing the path and then the URL separately. Still, an example would be helpful.

Comment (gilnoycloudinary) [reply examples/js/loaders/GLTFLoader.js]: Here is an example of a model - https://res.cloudinary.com/du46dspul/image/upload/f_gltf,fl_draco/gpm2qrjnowle2cvge8we

Comment (donmccurdy) [reply examples/js/loaders/GLTFLoader.js]: Ok thanks, this looks good to me. Here and below could you replace match code style with spaces inside parens and around if blocks?

```js
if ( /^https?:\/\//i.test( path ) && /^\//.test( url ) ) {

  path = path.replace( /(^https?:\/\/[^\/]+).*/i, '$1' );

}
```

Comment (gilnoycloudinary) [reply examples/js/loaders/GLTFLoader.js]: done

File: examples/js/loaders/GLTFLoader.js [hunk 2]
```diff
@@ -371,7 +371,8 @@ THREE.GLTFLoader = ( function () {
 
 		lightNode.name = lightDef.name || ( 'light_' + lightIndex );
 
-		return Promise.resolve( lightNode );
+		return Promise.
+		( lightNode );
```
Comment (donmccurdy) [inline examples/js/loaders/GLTFLoader.js]: typo here?

Comment (donmccurdy) [reply examples/js/loaders/GLTFLoader.js]: github's find/replace UI is weird this has happened to me before 😅

Comment (gilnoycloudinary) [reply examples/js/loaders/GLTFLoader.js]: strange as always  :-)

**Review by donmccurdy — APPROVED** [review]

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**